peregrine/tracking.py

```diff
@@ -227,6 +233,17 @@ def do_channel(chan, n=None, q_progress=None):
     progress = 0
     ms_tracked = 0
     i = 0
+    # For L2C, proceed in steps of full milliseconds up to the ms when
+    # handover succeeded. Do not set sample_index := chan.sample_index
+    # since the sub ms part of sample_index presents code phase.
+    # Therefore, skip just full ms steps to preserve code phase.
+    if chan.signal == 'l2c':
+      samples_per_ms = defaults.sampling_freq * defaults.code_period
```

Comment:
Let's use sampling_freq instead of defaults.sampling_freq here, becaue the parameter can be explicitly spefied with track() API call.